src/main/scala-2.11/markets/actors/participants/OrderIssuer.scala

```diff
@@ -15,41 +15,50 @@ limitations under the License.
 */
 package markets.actors.participants
 
-import markets.orders.Order
+
+import markets.actors.participants.strategies.OrderIssuingStrategy
+import markets.orders.{AskOrder, BidOrder}
 import markets.orders.limit.{LimitAskOrder, LimitBidOrder}
 import markets.orders.market.{MarketAskOrder, MarketBidOrder}
-import markets.actors.participants.strategies.TradingStrategy
 import markets.tradables.Tradable
 
 
 trait OrderIssuer {
   this: MarketParticipant =>
 
-  def tradingStrategy: TradingStrategy
+  def askOrderIssuingStrategy: OrderIssuingStrategy[AskOrder]
+
+  def bidOrderIssuingStrategy: OrderIssuingStrategy[BidOrder]
 
   def orderIssuerBehavior: Receive = {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Cyclomatic complexity of 11 exceeds max of 10](https://www.codacy.com/app/davidrpugh/markets-sandbox/file/2585426019/issues/source?bid=2921361&fileBranchId=3304063#l33)